SetupDataPkg/Tools/ConfigEditor.py

```diff
@@ -800,40 +800,36 @@ def load_from_delta(self):
     def load_delta_file(self, path):
         # assumption is there is only one yaml file (deprecated) but there may be multiple xml files
         # so we can only load this delta file if the file name matches to this xml data
-        file_id = -1
         yml_id = -1
-        xml_id = -1
+        updated_knobs = 0
         is_variable_list_format = True
         for idx in self.cfg_data_list:
             # if we have a yaml, fall back to old behavior
-            # if loading xml, ensure that the file path we have matches the csv
-            # we are trying to load
+            # if loading xml, ensure that knobs GUID + name exist in any loaded XML
             if self.cfg_data_list[idx].config_type == 'yml':
                 yml_id = idx
-            elif re.search(self.cfg_data_list[idx].cfg_data_obj._cur_page, path) is not None:
-                xml_id = idx
-
-        if xml_id == -1:
-            raise Exception('Associated XML not found! CSV filename must have the XML filename as a prefix')
+            else:
+                self.reload_config_data_from_bin(self.cfg_data_list[idx].org_cfg_data_bin, idx, is_variable_list_format)
+                try:
+                    updated_knobs += self.cfg_data_list[idx].cfg_data_obj.override_default_value(path)
+                except Exception as e:
+                    messagebox.showerror("LOADING ERROR", str(e))
+                    return
 
         if path.endswith('.dlt'):
-            file_id = yml_id
             is_variable_list_format = False
+            self.reload_config_data_from_bin(self.cfg_data_list[yml_id].org_cfg_data_bin, yml_id, is_variable_list_format)
+            try:
+                self.cfg_data_list[yml_id].cfg_data_obj.override_default_value(path)
+            except Exception as e:
+                messagebox.showerror("LOADING ERROR", str(e))
+                return
         elif path.endswith('.csv'):
-            file_id = xml_id
+            if updated_knobs == 0:
+                raise RuntimeError('Loaded CSV did not apply to any loaded config file!')

```

Comment:
Above you are using messagebox.showerror, but here you are raising a runtime error.

Would the updated_knobs == 0 be a recoverable error? Maybe they need to load a different config data bin? or is it a critical failure?

Reply:
Yeah, the existing codebase is very inconsistent and uses a mixture of exceptions and message boxes. The main pattern seems to be using exceptions. That being said, I can change this to a message box, no need to have it be a critical failure. Probably an issue to file in the repo to go and change all the other non-critical exceptions. However, with the upcoming scrub of the YAML stuff, a lot of that will go away.